tests/ratchet/test_ratchet_skipped_messages.c

```diff
@@ -0,0 +1,215 @@
+    vsc_buffer_destroy(&pub1);
+    vsc_buffer_destroy(&priv2);
+    vsc_buffer_destroy(&pub2);
+    vsc_buffer_destroy(&priv3);
+    vsc_buffer_destroy(&pub3);
+
+    vscr_ratchet_skipped_messages_destroy(&msgs);
+}
+
+void
+test__add_key__many_keys__keys_number_should_be_limited(void) {
+    vscr_ratchet_skipped_messages_t *msgs = vscr_ratchet_skipped_messages_new();
+
+    vsc_buffer_t *pub[vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES + 1];
+    vscr_ratchet_skipped_message_key_t *keys_arr[vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES + 1];
+
+    for (size_t i = 0; i < vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES; i++) {
+        vsc_buffer_t *priv;
+
+        generate_raw_keypair(&priv, &pub[i]);
+
+        vsc_buffer_destroy(&priv);
+
+        vscr_ratchet_skipped_message_key_t *key = vscr_ratchet_skipped_message_key_new();
+        key->message_key = vscr_ratchet_message_key_new();
+        key->message_key->index = (uint32_t)i;
+        memcpy(key->public_key, vsc_buffer_bytes(pub[i]), sizeof(key->public_key));
+        keys_arr[i] = key;
+
+        vscr_ratchet_skipped_messages_add_key(msgs, key);
+
+        for (size_t j = 0; j <= i; j++) {
+            TEST_ASSERT_EQUAL(keys_arr[j], vscr_ratchet_skipped_messages_find_key(msgs, j, vsc_buffer_data(pub[j])));
+        }
+    }
+
+    vsc_buffer_t *priv;
+
+    generate_raw_keypair(&priv, &pub[vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES]);
+
+    vsc_buffer_destroy(&priv);
+
+    vscr_ratchet_skipped_message_key_t *key = vscr_ratchet_skipped_message_key_new();
+    key->message_key = vscr_ratchet_message_key_new();
+    key->message_key->index = (uint32_t)vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES;
+    memcpy(key->public_key, vsc_buffer_bytes(pub[vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES]),
+            sizeof(key->public_key));
+    keys_arr[vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES] = key;
+
+    vscr_ratchet_skipped_messages_add_key(msgs, key);
+
+    TEST_ASSERT_EQUAL(NULL, vscr_ratchet_skipped_messages_find_key(msgs, 0, vsc_buffer_data(pub[0])));
+
+    for (size_t j = 1; j <= vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES; j++) {
+        TEST_ASSERT_EQUAL(keys_arr[j], vscr_ratchet_skipped_messages_find_key(msgs, j, vsc_buffer_data(pub[j])));
+    }
+
+    for (size_t i = 0; i < vscr_ratchet_common_hidden_MAX_SKIPPED_MESSAGES + 1; i++) {
+        vsc_buffer_destroy(&pub[i]);
+    }
+
+    vscr_ratchet_skipped_messages_destroy(&msgs);
+}
+
+#endif
+
+// --------------------------------------------------------------------------
+// Entrypoint.
+// --------------------------------------------------------------------------
+int
+main(void) {
+    UNITY_BEGIN();
+
+#if TEST_DEPENDENCIES_AVAILABLE
+    RUN_TEST(test__skipped_messages__adding_chains__should_be_correct);
+    RUN_TEST(test__add_key__many_keys__keys_number_should_be_limited);
+#else
+    RUN_TEST(test__nothing__feature_disabled__must_be_ignored);
+#endif
+
+    return UNITY_END();
+}
```

Comment:
Code style: ensures that there is always at lease 1 empty line at the end of the file.